File: src/main/java/eu/happybe/openapi/utils/PlayerUtils.java
package eu.happybe.openapi.utils;

import cn.nukkit.Player;
import eu.happybe.openapi.ranks.Rank;
import eu.happybe.openapi.ranks.RankDatabase;

public class PlayerUtils {

    public static void updateNameTag(Player player) {
        PlayerUtils.updateNameTag(player, player.namedTag.exist("NameTagColor") ? player.namedTag.getString("NameTagColor") : "§e");
    }

    public static void updateNameTag(Player player, String color) {
        Rank rank = RankDatabase.getPlayerRank(player);
        if(rank == null) {
            rank = RankDatabase.getRankByName("Guest");
        }

        player.namedTag.putString("NameTagColor", color);
        player.setNameTag(rank.getFormatForNameTag() + color + player.getName() + "\n§b" + DeviceData.getDeviceName(player));
    }
}

File: src/main/java/eu/happybe/openapi/utils/PlayerUtils.kt
package eu.happybe.openapi.utils

import cn.nukkit.Player
import eu.happybe.openapi.ranks.RankDatabase

object PlayerUtils {
    fun updateNameTag(player: Player) {
        updateNameTag(player, if (player.namedTag.exist("NameTagColor")) player.namedTag.getString("NameTagColor") else "§e")
    }

    fun updateNameTag(player: Player, color: String) {
        var rank = RankDatabase.getPlayerRank(player)
        if (rank == null) {
            rank = RankDatabase.getRankByName("Guest")
        }
        player.namedTag.putString("NameTagColor", color)
        player.nameTag = """
               ${rank.formatForNameTag}$color${player.name}
               §b${DeviceData.getDeviceName(player)}
               """.trimIndent()
    }
}